### ISSUE TITLE
Specify terminal group for integrated terminal in launch configuration

### ISSUE THREAD

**Issue by markfinlabs** [Microsoft/vscode#68123]

**Comment by markfinlabs** [comment]
A further comment: I guess perhaps the most logical thing would be to simply support the same ```presentation``` group of options as for tasks (applicable only when using integrated terminal)...? 

https://code.visualstudio.com/docs/editor/tasks#_output-behavior

**Comment by Tyriar** [comment]
I'm not sure we want this group concept proliferating to everything @alexr00? Split APIs for the terminal will likely look more like the view column APIs.

**Comment by markfinlabs** [comment]
Please when considering this realise how currently painful it is with compound launch configs that launch multiple integrated terminals. You are forever switching between terminals (whose order is a bit random). Anything that allows a compound launch config to launch integrated terminals in side-by-side clusters to group related terminals visually would do the job. Groups seemed the perfect solution but then I don't understand how the view column APIs work. Also fragmentation between the operation of integrated terminals for tasks and launch configurations doesn't seem good?

**Comment by markfinlabs** [comment]
Just to give my thoughts on the concepts and ideal roadmap here.

I think there is a need to support a dynamic grid of terminals that grows and shrinks as associated tasks or debug configs are started/stopped etc. That is for me what the whole group concept is about. I think the group concept is great and could easily (and usefully) be extended to be a 2D grid and support position ordering etc.

This group concept can be seen as *completely independent of and complementary to* the useful idea of a static UI grid of terminals which persists between sessions of VS Code and which could also ultimately be 2D or even 3D (stacks of 2D grids like we have stacks of 1D grids today). Each terminal in such a static arrangement could itself be a group - possibly empty and awaiting its first dynamic member (task or launch config starting).

Ultimate support for both static and dynamic 2D grids as described here seems to me a possible ultimate aspiration and would be very powerful indeed.

**Comment by weinand** [comment]
@markfinlabs Please be aware that in contrast to tasks, launch configurations do not know anything about terminals. If a debugger supports launching in the integrated terminal, this is an implementation detail of the debugger. And the launch config properties related to terminals are part of the debugger's JSON schema, not part of VS Code's schema.
So VS Code is not a position to introduce a "terminal group" in all launch configs.

**Comment by markfinlabs** [comment]
Thanks for the reply and the additional info. Would it make sense to expose the same 'presentation' property used by tasks as a runInTerminal argument when integrated terminal is selected? (Would helps unify use of integrated terminals between tasks and debug.)

**Comment by weinand** [comment]
A specified "presentation property" in the "runInTerminal" request makes only sense if the specification is "universal" (independent from VS Code). If it is very specific to VS Code, we do not want to include it in the DAP.
But in that case we could consider to support passing an opaque option bag via the "runInTerminal" request.

**Comment by markfinlabs** [comment]
Okay, got it, DAP is a general protocol which I didn't know much about which I guess allows people to create debug profiles that work with multiple IDEs. So I see why this has to be an IDE-specific option bag for enhanced experience with VS Code.

**Comment by markfinlabs** [comment]
Just to note, on StackOverflow, there is a desire to be able to clear the integrated terminal when starting debugging.
https://stackoverflow.com/questions/56619836/vscode-clear-integrated-terminal-when-debug-starts
I also need this functionality, particularly when using Jest for unit testing.
I believe the above proposal for support of presentation via an opaque option bag would solve this issue. 😀

**Comment by tommitytom** [comment]
Has there been any progress made on this?  I just started working with compound debugging and it seemed obvious that they would start in terminals side by side!  Is there a way to manually arrange the terminals once debugging has started?

**Comment by ldarshan1991** [comment]
> Has there been any progress made on this? I just started working with compound debugging and it seemed obvious that they would start in terminals side by side! Is there a way to manually arrange the terminals once debugging has started?

Is there a way to merge the 2 integrated terminals into one integrated terminal so that they split into two?
At least as a temporary workaround until a cleaner solution is provided.

**Comment by samuelms1** [comment]
Is there any update on this?  Like @tommitytom, I just got started on compound debugging and it would be really nice to have them split.  There doesn't appear to be a way to split them after they've started to run like @ldarshan1991 suggested.  Thanks for hearing me out.  :D

**Comment by roblourens** [comment]
That presentation group is not related to the terminal group. We still need DAP support for something like this.

**Comment by nbelakovski** [comment]
Just wanted to add that I've come up with something of a workaround for this using editor group and editor terminals. After I launch my compound debug and the new integrated terminals are created, I drag them out into the editor area to make them separate tabs, and then use View -> Editor Layout -> Grid 2x2 so that I can see both terminals side by side.

Would be great if they could launch directly into split panes, but this works for now.

**Comment by airstrike** [comment]
I was just digging into this as I start Advent of Code with ReScript which requires two scripts to run in parallel `npm run res:dev` and `npm run dev`. It seems to be achievable with vscode today, so sharing the below in case helpful.

I'm actually using `bun` instead of `npm` and it works fine too (just comment / uncomment the two lines accordingly)

```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "ReScript Watcher",
            // "type": "npm",
            // "script": "res:dev",
            "type": "shell",
            "command": "bun run res:dev",
            "isBackground": true,
            "problemMatcher": [],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "presentation": {
                "group": "dev",
            }
        },
        {
            "label": "Vite Server",
            // "type": "npm",
            // "script": "dev",
            "type": "shell",
            "command": "bun run dev",
            "isBackground": true,
            "problemMatcher": [],
            "group": "build",
            "presentation": {
                "group": "dev",
            }
        },
        {
            "label": "Run dev tasks",
            "dependsOn": [
                "ReScript Watcher",
                "Vite Server"
            ],
            "dependsOrder": "parallel",
            "problemMatcher": [],
            "presentation": {
                "group": "dev",
                "reveal": "always",
                "panel": "shared"
            }
        }
    ]
}
```

**Comment by leobaldoneto** [comment]
> I was just digging into this as I start Advent of Code with ReScript which requires two scripts to run in parallel `npm run res:dev` and `npm run dev`. It seems to be achievable with vscode today, so sharing the below in case helpful.
> 
> I'm actually using `bun` instead of `npm` and it works fine too (just comment / uncomment the two lines accordingly)
> 
> ```json
> {
>     "version": "2.0.0",
>     "tasks": [
>         {
>             "label": "ReScript Watcher",
>             // "type": "npm",
>             // "script": "res:dev",
>             "type": "shell",
>             "command": "bun run res:dev",
>             "isBackground": true,
>             "problemMatcher": [],
>             "group": {
>                 "kind": "build",
>                 "isDefault": true
>             },
>             "presentation": {
>                 "group": "dev",
>             }
>         },
>         {
>             "label": "Vite Server",
>             // "type": "npm",
>             // "script": "dev",
>             "type": "shell",
>             "command": "bun run dev",
>             "isBackground": true,
>             "problemMatcher": [],
>             "group": "build",
>             "presentation": {
>                 "group": "dev",
>             }
>         },
>         {
>             "label": "Run dev tasks",
>             "dependsOn": [
>                 "ReScript Watcher",
>                 "Vite Server"
>             ],
>             "dependsOrder": "parallel",
>             "problemMatcher": [],
>             "presentation": {
>                 "group": "dev",
>                 "reveal": "always",
>                 "panel": "shared"
>             }
>         }
>     ]
> }
> ```

It works with tasks.json, but don't with launch.json in configurations and compounds.

**Comment by ashikns** [comment]
Did anyone ever figure out a solution for this? I'm trying to launch a node-terminal and debugpy in split terminals.
